cms/io/rpc.py

```diff
@@ -478,13 +479,28 @@ def _connect(self):
 
         """
         try:
-            sock = gevent.socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            sock.connect(self.remote_address)
-        except OSError as error:
-            logger.debug("Couldn't connect to %s: %s.",
-                         self._repr_remote(), error)
-        else:
-            self.initialize(sock, self.remote_service_coord)
+            # Try to resolve the address, this can lead to many possible
+            # addresses, we'll try all of them.
+            addresses = gevent.socket.getaddrinfo(
+                self.remote_address.ip,
+                self.remote_address.port,
+                type=socket.SOCK_STREAM)
+        except socket.gaierror:
+            logger.warning("Cannot resolve %s.", self.remote_address)
+            raise
+
+        for family, *rest, sockaddr in addresses:
```

Comment:
Sorry, one more comment: you can get `type` and `proto` here and pass them to `socket()`, as recommended in the docs.